pennylane/qchem/tapering.py

```diff
@@ -616,65 +669,74 @@ def taper_operation(operation, generators, paulixops, paulix_sector, wire_order,
                 *qml.utils.decompose_hamiltonian(gen_mat, wire_order=wire_order, hide_identity=True)
             )
-            qml.simplify(gen_op)
-            if gen_op.ops[0].label() == qml.Identity(wires=[wire_order[0]]).label():
-                gen_op -= qml.Hamiltonian([gen_op.coeffs[0]], [qml.Identity(wires=wire_order[0])])
+            qml.simplify(op_gen)
+            if op_gen.ops[0].label() == qml.Identity(wires=[wire_order[0]]).label():
+                op_gen -= qml.Hamiltonian([op_gen.coeffs[0]], [qml.Identity(wires=wire_order[0])])
         else:  # Single-parameter gates
             try:
-                gen_op = qml.generator(operation, "hamiltonian")
+                op_gen = qml.generator(operation, "hamiltonian")
 
             except ValueError as exc:
                 raise NotImplementedError(
-                    f"Generator for {operation} is not implemented, please provide it with 'gen_op' args."
+                    f"Generator for {operation} is not implemented, please provide it with 'op_gen' args."
                 ) from exc
     else:  # check that user-provided generator is correct
-        if not isinstance(gen_op, qml.Hamiltonian):
+        if not isinstance(op_gen, qml.Hamiltonian):
             raise ValueError(
-                f"Generator for the operation needs to be a qml.Hamiltonian, but got {type(gen_op)}."
+                f"Generator for the operation needs to be a qml.Hamiltonian, but got {type(op_gen)}."
             )
         coeffs = 1.0
         if operation.parameters:
             coeffs = functools.reduce(lambda i, j: i * j, operation.parameters)
-        mat1 = scipy.linalg.expm(1j * qml.matrix(gen_op, wire_order=wire_order) * coeffs)
+        mat1 = scipy.linalg.expm(1j * qml.matrix(op_gen, wire_order=wire_order) * coeffs)
         mat2 = qml.matrix(operation, wire_order=wire_order)
         phase = np.divide(mat1, mat2, out=np.zeros_like(mat1, dtype=complex), where=mat1 != 0)[
             np.nonzero(np.round(mat1, 10))
         ]
         if not np.allclose(phase / phase[0], np.ones(len(phase))):  # check if the phase is global
             raise ValueError(
-                f"Given gen_op: {gen_op} doesn't seem to be the correct generator for the {operation}."
+                f"Given op_gen: {op_gen} doesn't seem to be the correct generator for the {operation}."
             )
 
     if np.all(
         [
             [
                 qml.is_commuting(op1, op2)
-                for op1, op2 in itertools.product(generator.ops, gen_op.ops)
+                for op1, op2 in itertools.product(generator.ops, op_gen.ops)
             ]
             for generator in generators
         ]
     ):
-        gen_tapered = qml.taper(gen_op, generators, paulixops, paulix_sector)
+        gen_tapered = qml.taper(op_gen, generators, paulixops, paulix_sector)
     else:
         gen_tapered = qml.Hamiltonian([], [])
     qml.simplify(gen_tapered)
 
+    def _tapered_op(params):
```

Comment:
This function needs complete docstring, including Args, Returns and Example.

Reply:
Since this is a private function, the full docstring conventions aren't needed :) Although providing more details on use can still be valuable to other developers